Tasks/Common/webdeployment-common/msdeployutility.ts

```diff
@@ -37,14 +37,26 @@ export function getMSDeployCmdArgs(webAppPackage: string, webAppName: string, pu
         msDeployCmdArgs += " -source:IisApp=\'" + webAppPackage + "\'";
         msDeployCmdArgs += " -dest:iisApp=\'" + webApplicationDeploymentPath + "\'";
     }
-    else {       
-        msDeployCmdArgs += " -source:package=\'" + webAppPackage + "\'";
-
-        if(isParamFilePresentInPacakge) {
-            msDeployCmdArgs += " -dest:auto";
-        }
-        else {
-            msDeployCmdArgs += " -dest:contentPath=\'" + webApplicationDeploymentPath + "\'";
+    else {
+        if (webAppPackage && webAppPackage.toLowerCase().endsWith('.war')) {
+            tl.debug('WAR: webAppPackage = ' + webAppPackage);
+            let warFile = path.basename(webAppPackage.toLowerCase().slice(0, webAppPackage.length - '.war'.length));
+            tl.debug('WAR: warFile = ' + warFile);
+            warFile = (virtualApplication) ? warFile + "/" + virtualApplication + '.war' : warFile + '.war';
+            tl.debug('WAR: warFile = ' + warFile);
+            msDeployCmdArgs += " -source:contentPath=\'" + webAppPackage + "\'";
+            // tomcat, jetty location on server => /site/webapps/
+            tl.debug('WAR: dest = /site/webapps/' + warFile);
+            msDeployCmdArgs += " -dest:contentPath=\'/site/webapps/" + warFile + "\'";
+        } else {
+            msDeployCmdArgs += " -source:package=\'" + webAppPackage + "\'";
+
+            if(isParamFilePresentInPacakge) {
+                msDeployCmdArgs += " -dest:auto";
+            }
+            else {
+                msDeployCmdArgs += " -dest:contentPath=\'" + webApplicationDeploymentPath + "\'";
+            }
```

Comment:
it would be great if one day this function returned an array since toolrunner handles escaping and the convention already set forth in this function is concat without escaping.